devices.go

```diff
@@ -46,14 +47,15 @@ func (d Device) String() string {
 
 // DeviceCreateRequest type used to create a Packet device
 type DeviceCreateRequest struct {
-	HostName     string   `json:"hostname"`
-	Plan         string   `json:"plan"`
-	Facility     string   `json:"facility"`
-	OS           string   `json:"operating_system"`
-	BillingCycle string   `json:"billing_cycle"`
-	ProjectID    string   `json:"project_id"`
-	UserData     string   `json:"userdata"`
-	Tags         []string `json:"tags"`
+	HostName      string   `json:"hostname"`
+	Plan          string   `json:"plan"`
+	Facility      string   `json:"facility"`
+	OS            string   `json:"operating_system"`
+	BillingCycle  string   `json:"billing_cycle"`
+	ProjectID     string   `json:"project_id"`
+	IPXEScriptUrl string   `json:"ipxe_script_url"`
```

Comment:
I think ipxe_script_url also needs to be `omitempty` or else userdata never works.  I'll push a fix.